framework/src/modules/bft/bft_params.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright © 2021 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import { codec } from '@liskhq/lisk-codec';
+import { BIG_ENDIAN, intToBuffer } from '@liskhq/lisk-cryptography';
+import { ImmutableSubStore } from '../../node/state_machine';
+import { SubStore } from '../../node/state_machine/types';
+import { BFTParameterNotFoundError } from './errors';
+import { BFTParameters, bftParametersSchema } from './schemas';
+
+export const getBFTParameters = async (
+	paramsStore: ImmutableSubStore,
+	height: number,
+): Promise<BFTParameters> => {
+	const start = intToBuffer(0, 4, BIG_ENDIAN);
+	const end = intToBuffer(height, 4, BIG_ENDIAN);
+	const results = await paramsStore.iterate({
+		limit: 1,
+		start,
+		end,
+		reverse: true,
+	});
+	if (results.length !== 1) {
+		throw new BFTParameterNotFoundError();
+	}
+	const [result] = results;
+	return codec.decode<BFTParameters>(bftParametersSchema, result.value);
+};
+
+export const deleteBFTParameters = async (paramsStore: SubStore, height: number): Promise<void> => {
+	const start = intToBuffer(0, 4, BIG_ENDIAN);
+	const end = intToBuffer(height, 4, BIG_ENDIAN);
+	const results = await paramsStore.iterate({
+		start,
+		end,
+	});
+	if (results.length <= 1) {
```

Comment:
Cosmetic comment: I think this check is not necessary as the loop will not be executed if `results.length == 1`.

Reply:
Maybe we can also add a comment below as clarification what this function does, e.g.

> delete all BFT Parameters except the one of largest height which is at most the input height

Reply:
the equal case might be not necessary, but empty case is important, so ill keep the condition and add the comment

Reply:
Note that also for empty results array the loop is not executed, but it is also correct like this.